src/main/groovy/com/twcable/grabbit/client/servlets/GrabbitJobServlet.groovy

```diff
@@ -147,6 +147,28 @@ class GrabbitJobServlet extends SlingAllMethodsServlet {
         response.writer.write(new JsonBuilder(jobExecutionIds).toString())
     }
 
+    @Override
+    protected void doPost(SlingHttpServletRequest request, SlingHttpServletResponse response)  {
+        String jobExecutionId = request.getParameter("jobId") ?: ""
+        if(!jobExecutionId.isLong()) {
+            log.warn "Parameter ${jobExecutionId} 'jobId' is invalid"
+            response.status = SC_BAD_REQUEST
+            response.writer.write("Parameter 'jobId' is invalid")
+            return
+        }
+        try {
+            SimpleJobOperator jobOperator = configurableApplicationContext.getBean(SimpleJobOperator)
+            jobOperator.stop(jobExecutionId.toLong())
```

Comment:
we should also look to see whether any other cleanup of like the inputstream / session is needed after stopping the Job.

Take a look at `ClientBatchJobExecutionListener.afterJob()` 

Also, this will only stop the client .. server will still be wasting time and resources in sending his content over :-(